src/test/simulators/simulatorAgileService.ts

```diff
@@ -0,0 +1,43 @@
+import { BacklogItem, Project, Sprint } from "../../models";
+import { BaseAgileService } from "../../services";
+import { ModelSimulator } from "./modelSimulator";
+
+interface MockAgileServiceFunctions {
```

Comment:
Why do we need a Mock interface here? Can't we mock a `SimulatorAgileService` with some mocking library and have it discover and automatically do what we're writing here?

Reply:
This just allows us to expose internal functions to the constructor so we can make assertions on how, when or if these functions are called. [Here is an example](https://github.com/tbarlow12/cse-cli/pull/29/files#diff-a6bb3fb434804ef0978d990b6ace9aa2af6a7e7c99b190dd17310870f4d9a623R25-R30). I'm open to other ways of doing it too though 👍